buildmodel/dockerversions/dockerversions.go

```diff
@@ -85,13 +85,24 @@ func (a ArchEnv) GoImageArchSuffix() string {
 	return ""
 }
 
-// GoImageOSArchKey creates the string used by docker-library/golang to identify this OS/arch in its
-// versions.json file. This is upstream behavior we are conforming to.
+// GoImageOSArchKey replicates upstream behavior, creating the string used by docker-library/golang
+// to identify this OS/arch in its versions.json file. Linux is the default OS, so if it is the
+// ArchEnv's OS, it isn't included in this string.
 func (a ArchEnv) GoImageOSArchKey() string {
-	var s = a.GOARCH + a.GoImageArchSuffix()
-	// OSes other than Linux need a prefix.
+	var s = a.GoImageArchKey()
 	if a.GOOS != "linux" {
 		s = a.GOOS + "-" + s
 	}
 	return s
 }
+
+// GoImageArchKey replicates upstream behavior, creating the arch string used by
+// docker-library/golang and .NET Docker to identify this arch in the versions.json and
+// manifest.json files.
+func (a ArchEnv) GoImageArchKey() string {
```

Comment:
Curious: is there a reason you're passing by value instead of pointer here? It's now a triple word struct which is my general line for "maybe start passing via ref".

Reply:
It looks like three string fields -> three sort-of-slices ([pointer + length, no cap](https://github.com/golang/go/blob/91b9915d3f6f8cd2e9e9fda63f67772803adfa03/src/runtime/string.go#L238-L241)) -> 6 words.

(Worked backwards from `unsafe.Sizeof` and noticed that `[]rune` is 3 words but `string` is 2. 🙂)

No reason to keep it as a value receiver here--changing.

Reply:
FWIW, these guidelines look good to me: https://github.com/golang/go/wiki/CodeReviewComments#receiver-type. There are *very* few reasons there to use a value receiver: if the receiver type is a map/func/chan, or if it's tiny and demonstrably helps perf.